File: set2_solutions.py
# Set 2: Functions & Loops

# 1. Function greet
def greet(name):
    return f"Hello, {name}!"

print(greet("Iradi"))

# 2. Function add
def add(num1, num2):
    return num1 + num2

print(add(5, 10))

# 3. Function isEven
def isEven(number):
    return number % 2 == 0

print(isEven(4))

# 4. For loop to print numbers from 1 to 10
for i in range(1, 11):
    print(i)

# 5. While loop to print the first 5 multiples of 3
i = 1
while i <= 5:
    print(i * 3)
    i += 1

# 6. For loop to calculate factorial of n
def factorial(n):
    result = 1
    for i in range(1, n + 1):
        result *= i
    return result

print(factorial(5))

File: set2_solutions.js
// 1. Greet a user
function greet(name) {
  return "Hello, " + name + "!";
}
console.log(greet("Iradi"));

// 2. Add two numbers
function add(a, b) {
  return a + b;
}
console.log("Sum: " + add(5, 10));

// 3. Check if a number is even
function isEven(number) {
  return number % 2 === 0;
}
console.log("Is 4 even? " + isEven(4));

// 4. Print numbers from 1 to 10 using a for loop
for (let i = 1; i <= 10; i++) {
  console.log(i);
}

// 5. Print the first 5 multiples of 3 using a while loop
let count = 1;
while (count <= 5) {
  console.log(count * 3);
  count++;
}

// 6. Calculate the factorial of a number
function factorial(n) {
  let result = 1;
  for (let i = 1; i <= n; i++) {
    result *= i;
  }
  return result;
}
console.log("Factorial of 5: " + factorial(5));
